Code/Tools/ProjectManager/Source/CreateAGemScreen.h

```diff
@@ -0,0 +1,131 @@
+#include <QTabBar>
+#include <QTabWidget>
+#include <QDialogButtonBox>
+#include <QPushButton>
+#include <QScrollArea>
+
+
+#endif
+
+class TabBar : public QTabBar
+{
+public:
+    QSize tabSizeHint(int index) const
+    {
+        QSize s = QTabBar::tabSizeHint(index);
+        s.transpose();
+        return s;
+    }
+
+protected:
+    void paintEvent(QPaintEvent* /*event*/)
+    {
+        QStylePainter painter(this);
+        QStyleOptionTab opt;
+
+        QPoint currentTopPosition(115, 175);
+
+        QStringList strs = { "1. Gem Setup", "2. Gem Details", "3. Creator Details" };
+        for (int i = 0; i < count(); i++)
+        {
+            initStyleOption(&opt, i);
+            painter.drawControl(QStyle::CE_TabBarTabShape, opt);
+            painter.save();
+            QSize s = opt.rect.size();
+            s.transpose();
+            QRect r(QPoint(), s);
+            r.moveCenter(opt.rect.center());
+            opt.rect = r;
+            QPoint c = tabRect(i).center();
+            QPoint leftJustify(currentTopPosition);
+            leftJustify.setX(30 + (int)(0.5 * opt.rect.width()));
+            painter.translate(leftJustify);
+            currentTopPosition.setY(currentTopPosition.y() + 55);
+            //painter.rotate(90);
+            painter.translate(-c);
+            // painter.drawControl(QStyle::CE_TabBarTabLabel, opt);
+            painter.drawItemText(r, Qt::AlignLeft, QApplication::palette(), true, strs.at(i));
+            painter.restore();
+        }
+    }
+};
+
+class TabWidget : public QTabWidget
+{
+public:
+    TabWidget(QWidget* parent = 0)
+        : QTabWidget(parent)
+    {
+        setTabBar(new TabBar);
+        setTabPosition(QTabWidget::West);
+    }
+};
+
+
+namespace O3DE::ProjectManager
+{
+
+    class CreateAGemScreen : public ScreenWidget
+    {
+    public:
+        explicit CreateAGemScreen(QWidget* parent = nullptr);
+        ~CreateAGemScreen() = default;
+
+    public slots:
+
+    protected:
+
+    private slots:
+        void OnGemDisplayNameUpdated();
+        void OnGemSystemNameUpdated();
+        void OnLicenseNameUpdated();
+        void OnCreatorNameUpdated();
+        void OnRepositoryURLUpdated();
+        void HandleBackButton();
+        void HandleNextButton();
+        void ConvertNextButtonToCreate();
+
+    private:
+        QScrollArea* CreateFirstScreen();
+        QScrollArea* CreateSecondScreen();
+        QScrollArea* CreateThirdScreen();
+        bool ValidateGemDisplayName();
+        bool ValidateGemSystemName();
+        bool ValidateLicenseName();
+        bool ValidateCreatorName();
+        bool ValidateRepositoryURL();
+        int m_numNextClicks = 0;
+
+        FormLineEditWidget* m_gemDisplayName;
+        FormLineEditWidget* m_gemSystemName;
+        FormLineEditWidget* m_license;
+        FormLineEditWidget* m_creatorName;
+        FormLineEditWidget* m_repositoryURL;
+
+        TabWidget* m_tabWidget;
+
+        QDialogButtonBox* m_backNextButtons;
```

Comment:
for consistency please init these to nullptr like the back and next buttons.

Reply:
Done.